src/main/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -188,7 +188,10 @@ public void addMetaDataLine(final VCFHeaderLine headerLine) {
      * @return all of the VCF header lines of the ##contig form in order, or an empty list if none were present
      */
     public List<VCFContigHeaderLine> getContigLines() {
-        return Collections.unmodifiableList(contigMetaData);
+        List<VCFContigHeaderLine> contigLines = new ArrayList<>(contigMetaData.size());
+        // this must preserve input order
+        contigLines.addAll(contigMetaData.values());
+        return Collections.unmodifiableList(contigLines);
```

Comment:
I don't agree with you here.  The return type is a List because we guarantee the return value is ordered.  In general we want to make outputs narrowly typed and inputs broadly typed since that makes our functions more useful.  There's a competing goal of hiding the implementation details so we're free to change them.  In general we probably want to return the narrowest abstract type that provides the guarantees we need for our method to function.  Enforcing conventions in javadoc is a much weaker and more brittle way to do things.